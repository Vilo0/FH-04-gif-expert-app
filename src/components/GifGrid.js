import React, { useEffect, useState } from 'react';
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {
    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick&limit=10&api_key=Ult6O0L8EpsXouxqN8xgxI6hyOsLgVjf';
        const resp = await fetch(url);
        const { data } = await resp.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url,
            };
        });

        console.log(gifs);
        setImages(gifs);
    }

    return (
        <>
            <h3>{ category }</h3>
            <div className='card-grid'>
                {
                    images.map( image => (
                        <GifGridItem key={image.id} { ...image } />
                    ))
                }
            </div>
        </>
    )
}

export default GifGrid
