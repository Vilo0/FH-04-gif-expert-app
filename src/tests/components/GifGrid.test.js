import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('prueban en el <GifGrid />', () => {
    const category = 'Luffy';

    test('debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    })
    
    test('debe de mostrar item cuand se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id: 'abc',
            title: 'cualquier cosa',
            url: 'https://localhost.jpg',
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

    })
    
})
