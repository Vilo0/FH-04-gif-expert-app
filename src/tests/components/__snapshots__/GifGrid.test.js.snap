// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueban en el <GifGrid /> debe de mostrar item cuand se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Luffy
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https://localhost.jpg"
    />
  </div>
</Fragment>
`;

exports[`prueban en el <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Luffy
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
